scripts/cc-cli/src/commands/registerAddress.ts

```diff
@@ -91,3 +91,94 @@ function errorMsg(s: string) {
 export function isValidPrivateKey(pk: string): boolean {
   return utils.isHexString(pk, 32);
 }
+
+export async function initCallerEthWallet(
+  options: OptionValues,
+): Promise<Wallet> {
+  try {
+    return await initWalletFromEnvOrPrompt("ETH_PRIVATE_KEY", options);
+  } catch (e) {
+    console.error(getErrorMessage(e));
+    process.exit(1);
+  }
+}
+
+async function initWalletFromEnvOrPrompt(
+  envVar: string,
+  options: OptionValues,
+): Promise<Wallet> {
+  const interactive = options.input;
+  const useMnemonic = options.ethMnemonic;
+  const inputName = useMnemonic ? "mnemonic" : "private key";
+  const generateWallet = useMnemonic
+    ? newWalletFromMnemonic
+    : newWalletFromPrivateKey;
+  const validateInput = useMnemonic ? isMnemonicValid : isValidPrivateKey;
+
+  if (!interactive && process.env[envVar] === undefined) {
+    throw new Error(
+      "Error: Must specify a private key using the environment variable ETH_PRIVATE_KEY or an interactive prompt",
+    );
+  }
+
+  if (process.env[envVar] !== undefined) {
+    const seed = process.env[envVar] as string;
+
+    if (!validateInput(seed)) {
+      throw new Error("Error: Private key is invalid");
+    }
+
+    return generateWallet(seed);
+  } else if (interactive) {
+    const promptResult = await prompts([
+      {
+        type: "password",
+        name: "seed",
+        message: `Specify the ${inputName} for the ethereum address`,
+        validate: validateInput,
+      },
+    ]);
+
+    const seed = promptResult.seed;
+
+    if (!seed) {
+      throw new Error("The mnemonic could not be retrieved from the prompt");
+    }
+
+    return generateWallet(seed);
+  }
+
+  throw new Error("The ethereum wallet could not be");
+}
+
+function newWalletFromMnemonic(mnemonic: string): Wallet {
+  let wallet: Wallet;
+
+  try {
+    wallet = Wallet.fromMnemonic(mnemonic);
+  } catch (e) {
+    throw new Error(
+      `Error: Could not create wallet from mnemonic: ${getErrorMessage(e)}`,
+    );
+  }
+
+  return wallet;
+}
+
+function newWalletFromPrivateKey(pk: string): Wallet {
+  let wallet: Wallet;
+
+  try {
+    wallet = new Wallet(pk);
+  } catch (e) {
+    throw new Error(
+      `Error: Could not create wallet from private key: ${getErrorMessage(e)}`,
+    );
+  }
```

Comment:
Both `newWalletFromPrivateKey` and `newWalletFromMnemonic` can benefit from quick positive and negative unit tests just to make sure they return a wallet or they throw an error if something goes wrong. (e.g. pass private key where mnemonic is expected and vice versa for the negative scenario).